### ISSUE TITLE
[Help] Prevent HOC update on navigation animation

### ISSUE THREAD

**Issue by SebT** [reactjs/react-redux#404]
Greetings,

**TL;DR** : How to prevent the `connect`'s `select` parameter function to be called on page navigation animation ?

I've been using react with redux and react-router for a while now and I've always faced an issue that is a bit annoying.

Let's say I have an `article` reducer in my store that is the article the user is currently viewing. So it's basically a simple json object with author name, content, title, etc.

So I make a High Order Component to display the article from the store (given via react-redux's `connect` function). The render method looks like :

```javascript
// ArticlePage.js
render() {
  return (
    <article>
      <h2>{this.props.article.title}</h2>
      <div>{this.props.article.content}</div>
      <button 
onClick={/*Call action to reset the current article (set to null in the store)*/}>Go back</button>
    </article>
  );
}
```
I use a navigation not from the URL (memory history) so it is impossible for the user to display the page when `article` is null in the store.

I use `ReactCSSTransitionGroup` to add an animation on page navigation. When I click on the "Go back" button, it launches a page transition and sets the `article` to null in the store. 

So, for the duration of my navigation animation, `this.props.article` is null and I get errors from my render method.

How to prevent that? The most obvious solution I see would be to prevent the `select` function to be called during transition, so that the component can't be updated on transition, but how can I achieve that?

**Comment by markerikson** [comment]
This is a usage question, and should be asked on Stack Overflow instead.

That said, the basic answer would be to either have your `mapState` function return an actual value instead of null, or update your rendering logic to either provide a default value, or just check if `article` is null and not render those fields.
